### ISSUE TITLE
How to use a stack these ConvLSTM layers?

### ISSUE THREAD

**Issue by giridhar13** [swathikirans/violence-recognition-pytorch#9]
I would like to pass the output of a VGG16 feature extractor (-1, 512, 7, 7) onto 2 sequential-identical convLSTM layers say layers with hidden state (7 *7 *256). Could you let me know how to pass the states between each layers?

A sample code for 1 layer of ConvLSTM is attached here 

class deepSignal(nn.Module):

    def __init__(self, mem_size):
        super(deepSignal, self).__init__()
        self.mem_size = mem_size
        self.convNet = model.vgg_attention.vgg16()
        self.conv_lstm = ConvLSTMCell(512, self.mem_size)-----------update here
        self.relu = nn.ReLU()
        self.maxpool = nn.MaxPool2d(2)
        self.classifier = nn.Sequential(nn.Linear(3 * 3 * self.mem_size, 1000), nn.ReLU(),
                                        nn.Linear(1000, 256), self.relu,
                                        nn.Linear(256, 2))
        # I have removed the batch norm in the classifier layer--look at the original code and add it

    def forward(self, x):
        state = None
        seqLen = x.size(0) - 1
        for t in range(0, seqLen):
            x1 = x[t] - x[t + 1]
            x1 = self.convNet(x1)
            state = self.conv_lstm(x1, state)----------------------updates here 
        x = self.maxpool(state[0])
        x = self.classifier(x.view(x.size(0), -1))
        return x

**Comment by swathikirans** [comment]
You can apply the hidden state of the first CLSTM cell as input to the second CLSTM cell. Please see the code below:

```
def __init__(self, mem_size):
    super(deepSignal, self).__init__()
    self.mem_size = mem_size
    self.convNet = model.vgg_attention.vgg16()
    ############################################################
    self.conv_lstm1 = ConvLSTMCell(512, self.mem_size)-----------update here
    self.conv_lstm2 = ConvLSTMCell(512, self.mem_size)
    ############################################################
    self.relu = nn.ReLU()
    self.maxpool = nn.MaxPool2d(2)
    self.classifier = nn.Sequential(nn.Linear(3 * 3 * self.mem_size, 1000), nn.ReLU(),
                                    nn.Linear(1000, 256), self.relu,
                                    nn.Linear(256, 2))
    # I have removed the batch norm in the classifier layer--look at the original code and add it

def forward(self, x):
    ############################################################
    state1 = None
    state2 = None
    ############################################################
    seqLen = x.size(0) - 1
    for t in range(0, seqLen):
        x1 = x[t] - x[t + 1]
        x1 = self.convNet(x1)
    ############################################################
        state1 = self.conv_lstm1(x1, state1)----------------------updates here 
        state2 = self.conv_lstm2(state1[0], state2)
    x = self.maxpool(state2[0])
    ############################################################
    x = self.classifier(x.view(x.size(0), -1))
    return x
```
